sqlglot/generator.py

```diff
@@ -552,6 +556,15 @@ def create_sql(self, expression: exp.Create) -> str:
                 )
             index_sql = "".join(indexes_sql)
 
+        postcreate_props_sql = ""
+        if properties and properties_locs[exp.Properties.Location.POST_CREATE]:
```

Comment:
can you refactor all the places where we do if properties and properties[xxx] and make properties default to a dict

properties = expression.args.get("Propertyers") or {}

then you can just use get